### ISSUE TITLE
Fepois: Add support for `predict` with `newdata` argument

### ISSUE THREAD

**Issue by s3alfisc** [py-econometrics/pyfixest#468]
## Context

Currently, the `newdata` argument is disabled for `Fepois.predict`. Unfortunately, I do not recall why... 😅 

## To do

- Enable the `newdata` argument for `Fepois.predict(). 
- Clean up ValueError when trying to run Fepois.predict() with `newdata` in `tests/test_errors.py`.
- Add unit tests to `tests/test_vs_r::test_single_fit`.

**Comment by leostimpfle** [comment]
Having had a look at #467 I think both this issue and #467 can be resolved in a single update.

I think it would be easiest to generalise the `predict` method of `feols` to always produce the linear predictor (`link`) and only handle the conversion to the non-linear predictor (`response`) in `fepois`.

**Comment by s3alfisc** [comment]
Sounds like a good plan! I agree, it makes sense to only define the `response` prediction in `Fepois`. Should I then assign this issue to you as well?

**Comment by leostimpfle** [comment]
Yes, happy to take it on!

**Comment by s3alfisc** [comment]
This can be close right @leostimpfle ?

**Comment by s3alfisc** [comment]
Ah no, actually not. Still throws an error. But in principle we should be able to support it now I think

**Comment by leostimpfle** [comment]
Yes, it should work without fixed effects. After your recent bugfix in the calculation of fixed effects, it will hopefully straightforward to support predict with both newdata and fixed effects. I should have some time this weekend to look into it
